src/app/legacy/providers/redeem-provider.tsx

```diff
@@ -74,6 +75,45 @@ export function RedeemProvider(props: { children: any }) {
     [inputToken, inputValue],
   )
 
+  const { data: quoteResult, isFetching: isFetchingQuote } = useQuery({
+    initialData,
+    queryKey: [
+      queryKey,
+      {
+        address,
+        inputToken,
+        inputTokenAmount: inputTokenAmount.toString(),
+        nativeTokenPrice,
+        provider,
+        publicClient,
+      },
+    ],
+    queryFn: async () => {
+      if (!address) return null
+      if (!provider || !publicClient) return null
+      if (inputTokenAmount <= 0) return null
+      const gasPrice = await provider.getGasPrice()
+      const legacyQuote = await getLegacyRedemptionQuote(
+        {
+          account: address,
+          inputTokenAmount,
+          inputToken,
+          gasPrice,
+          nativeTokenPrice,
+          slippage: 0,
+        },
+        publicClient,
+      )
+      return {
```

Comment:
probably it doesn't apply here so just saying, but you can separate fetching logic from data transformation logic if you want by returning the promise of whatever

and doing

```typescript
useQuery({
 select: (data: isTheTypeYouReturn) => ({...transform this however you want})
})
```